### PR TITLE
AC Automated Test - LRAC-11836 | master

### PR DESCRIPTION
**Pull request by ViniLopes20**
https://issues.liferay.com/browse/LRAC-11835 (Automation Ticket)
https://issues.liferay.com/browse/LRAC-11836 (Automation Ticket)
[Testray Result](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=1957453359&testrayRunId=1957453573)
[Story](https://issues.liferay.com/browse/LRAC-11663)
[Test Plan](https://docs.google.com/document/d/1lsYx9TBAgRxzkn4HZwe-NS6mtAvvY8WpXKmtqEDwUyo/edit#)

**Resolution:**

- I made a change in the `GENERIC_TEXT` and `GENERIC_TEXT_ORDER` paths to replace **`text()`** with **`normalize-space()`**, with this change we will be able to avoid formatting problems such as line breaks in the front text. This change will not change the use of the path, the behavior will remain the same, but now we will be able to use it in more cases;

- I moved the `SYNC_ALL_CONTACTS_TOGGLE` path to `ACUtils.path` and renamed it to `TOGGLE_ICON`, I made this change because the path was very generic and could be used for any Toggle icon so its name was very specific;

- I created `selectGenericOption` macro to be used for any use where we need to select options in a dropdown field;

- Other changes were new macros, paths and tests for the Email Report;

- In the `CheckEmailReportsAppearsForAllUsers` test I made a change in the automation regarding the test plan steps. In the test plan, it would only be for us to verify that the widget icon appears for any user (Scope of the first story), but as the second story was already placed inside the master code, I thought it best to verify that all users could perform the configuration to receive the email report;

### CODE REVIEW

**Comment by ViniLopes20** [comment]
ci:test:sf

**Review by christianstokes — APPROVED** [review]
Thank you @ViniLopes20 this looks really good and I like the refactors you did.

I will approve, but l leave it open for now so someone else can also review and double check ( @joshitagaki @calvinctvo @manuferreira ), but after getting some other eyes on this PR, feel free to forward it yourself @ViniLopes20. Thank you!

**Review by manuferreira — APPROVED** [review]
Thanks for the changes and for the tests too, Vini. :tada: :clap:

**Review by joshitagaki — APPROVED** [review]
Great job! LGTM! With the changes to the test CheckEmailReportsAppearsForAllUsers. Do you think it necessary to have the test [CanSaveEmailFrequencyReportDaily](https://issues.liferay.com/browse/LRAC-11841)? Since it would be a very similar.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/EmailReport.testcase [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+			ACProperties.addNewProperty(propertyName = "${propertyName}");
+		}
+
+		task ("Check that the Email Reports field appears and has the status disabled") {
+			ACUtils.viewGenericText(textValueList = "Email Reports: Disabled");
+		}
+
+		task ("Check that the Email Reports widget icon is disabled if the property is not synced") {
+			AssertVisible(
```
Comment (calvinctvo) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/EmailReport.testcase:106]: Interesting way to check for disabled button!

**Comment by ViniLopes20** [comment]
> Great job! LGTM! With the changes to the test CheckEmailReportsAppearsForAllUsers. Do you think it necessary to have the test [CanSaveEmailFrequencyReportDaily](https://issues.liferay.com/browse/LRAC-11841)? Since it would be very similar.

Hey @joshitagaki, I think this test uses the same steps as it would for `CanSaveEmailFrequencyReportDaily`, but it takes more steps. I think the `CanSaveEmailFrequencyReportDaily` would be a smoker test to verify that it can change the frequency to Daily, I think we can keep this test still and perform single automation for the **Daily**, **Monthly** and **Weekly**.

What do you think?

**Comment by ViniLopes20** [comment]
Hi team, Thanks for reviewing!

I'm forwarding it!

**Comment by ViniLopes20** [comment]
ci:forward

**Comment by joshitagaki** [comment]
> > Great job! LGTM! With the changes to the test CheckEmailReportsAppearsForAllUsers. Do you think it necessary to have the test [CanSaveEmailFrequencyReportDaily](https://issues.liferay.com/browse/LRAC-11841)? Since it would be very similar.
> 
> Hey @joshitagaki, I think this test uses the same steps as it would for `CanSaveEmailFrequencyReportDaily`, but it takes more steps. I think the `CanSaveEmailFrequencyReportDaily` would be a smoker test to verify that it can change the frequency to Daily, I think we can keep this test still and perform single automation for the **Daily**, **Monthly** and **Weekly**.
> 
> What do you think?

Yea I think it's fine to keep it. The only difference would be that it's not changing between different users with different permissions

**Pull request closed without merging**